tests/pytorch/test_config_validator_pytorch.py

```diff
@@ -0,0 +1,43 @@
+# Future
+from __future__ import print_function
+
+
+def test_supported_pytorch_version():
+    import smdebug.pytorch.singleton_utils
+    from unittest.mock import patch
+
+    with patch(
+        "smdebug.core.config_validator.is_framework_version_supported"
+    ) as override_is_current_version_supported:
+        smdebug.pytorch.singleton_utils.del_hook()
+
+        override_is_current_version_supported.return_value = False
+        hook = smdebug.pytorch.singleton_utils.get_hook()
+        assert hook == None
+    # Disengaging the hook also sets the environment variable USE_SMDEBUG to False, we would need to reset this
+    # variable for further tests.
+    import os
+    from smdebug.core.config_validator import reset_config_validator
+
+    del os.environ["USE_SMDEBUG"]
+    reset_config_validator()
+
+
+def test_disabling_detailed_profiler(simple_profiler_config_parser):
```

Comment:
There's a lot of cleanup and code redundancy for what looks like two straightforward tests. My suggestion:

First, set the SMP env variable string as a variable to be parametrized:
```
@pytest.mark.parametrize(smp_config, ['{"mp_parameters":{"partitions": 2}}', "{}"])
def test_disabling_detailed_profiler(smp_config, simple_profiler_config_parser):
    os.environ["SM_HPS"] = smp_config
    ...
```

Second, move the cleanup code outside of the test and the previous test. You can set up a really simple fixture to handle cleanup like this:
```
@pytest.fixture(autouse=True)
def cleanup():
    smdebug.pytorch.singleton_utils.del_hook()
    yield
    os.environ.pop("USE_SMDEBUG", None)
    os.environ.pop("SM_HPS", None)
    reset_config_validator()
    
```

Reply:
Linking more resources in case you'd like more context:

https://docs.pytest.org/en/6.2.x/parametrize.html
https://docs.pytest.org/en/6.2.x/fixture.html#autouse-fixtures-fixtures-you-don-t-have-to-request
https://docs.pytest.org/en/latest/how-to/fixtures.html?highlight=cleanup#yield-fixtures-recommended

Reply:
Done.